components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -10517,49 +10517,118 @@ private void addAPIPolicy(APIPolicy policy, Connection conn) throws SQLException
     /**
      * Update a API level throttling policy to database.
      * <p>
-     * policy will be inserted with that policy Id.
+     * If condition group already exists for the policy, that condition Group will be deleted and condition Group will
+     * be inserted to the database with old POLICY_ID.
      * </p>
      *
      * @param policy policy object defining the throttle policy
-     * @throws SQLException
+     * @throws APIManagementException
      */
-    private void updateAPIPolicy(APIPolicy policy, Connection conn) throws SQLException {
-        ResultSet resultSet = null;
-        PreparedStatement policyStatement = null;
-        String addQuery = SQLConstants.ThrottleSQLConstants.INSERT_API_POLICY_WITH_ID_SQL;
-        int policyId = policy.getPolicyId();
-
+    public APIPolicy updateAPIPolicy(APIPolicy policy) throws APIManagementException {
+        Connection connection = null;
+        PreparedStatement updateStatement = null;
+        PreparedStatement deleteStatement = null ;
+        PreparedStatement selectStatement = null;
+        String updateQuery;
+        int policyId = 0;
         try {
```

Comment:
Lets use try with resources statement for creating Connection, PreparedStatement and ResultSet objects so that they will be auto closed on exit, without the need to manage them manually.

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9211/commits/cea946a138945713c9af8c793cc218167525382f